### PR TITLE
fix(validation): allow https pushCallback URLs that contain :443

### PR DESCRIPTION
**Pull request by shane-tomlinson**
@vladikoff - here's a slightly different take on your PR. I added a `pushUrl` validator to avoid other https URLs being allowed to have a `:443`.

### CODE REVIEW

File: test/remote/device_tests.js [hunk 1]
```diff
@@ -378,7 +378,7 @@ describe('remote device', function () {
   it(
     'update device works with callback urls that have ports',
     () => {
-      var goodPushCallback = 'https://updates.push.services.mozilla.com:433'
+      var goodPushCallback = 'https://updates.push.services.mozilla.com:4430/';
```
Comment (vladikoff) [inline test/remote/device_tests.js]: @shane-tomlinson we need to keep a test that has just the `:443`

Comment (shane-tomlinson) [reply test/remote/device_tests.js]: Ah nooo, that was a mistake.

File: test/remote/device_tests.js [hunk 2]
```diff
@@ -376,9 +376,47 @@ describe('remote device', function () {
   )
 
   it(
-    'update device works with callback urls that have ports',
+    'update device works with callback urls that :443 as a port',
     () => {
-      var goodPushCallback = 'https://updates.push.services.mozilla.com:433'
+      var goodPushCallback = 'https://updates.push.services.mozilla.com:443'
```
Comment (shane-tomlinson) [inline test/remote/device_tests.js]: Should we add a test w/ a trailing slash too?

File: lib/routes/validators.js [hunk 3]
```diff
@@ -151,6 +151,31 @@ module.exports.url = function url(options) {
...
+module.exports.pushCallbackUrl = function pushUrl(options) {
+  const validator = isA.string().uri(options)
+  validator._tests.push(
+    {
+      func: (value, state, options) => {
+        if (value !== undefined && value !== null) {
+          let normalizedValue = value;
+          // Fx Desktop registers https push urls with a :443 which causes `isValidUrl`
+          // to fail because the :443 is expected to be normalized away.
+          if (/^https:\/\//.test(value) && /:443($|\/[^\d])/.test(value)) {
```
Comment (vbudhram) [inline lib/routes/validators.js]: Possible to combine this? Maybe something like this?

```javascript
/^(https:\/\/)[a-zA-Z0-9.-]*(:443)($|\/[^\d])/.test(value)
```

(Although now I see that might be harder to read)

File: test/remote/device_tests.js (hunk 2 above)
Comment (vladikoff) [reply test/remote/device_tests.js]: eh, might be too much

Comment (vladikoff) [reply test/remote/device_tests.js]: let's use this real callback here:

```
https://updates.push.services.mozilla.com:443/wpush/v1/gAAAAABbkq0Eafe6IANS4OV3pmoQ5Z8AhqFSGKtozz5FIvu0CfrTGmcv07CYziPaysTv_9dgisB0yr3UjEIlGEyoprRFX1WU5VA4nG-9tofPdA3FYREPf6xh3JL1qBhTa9mEFS2dSn--
```

**Comment by shane-tomlinson** [comment]
@jens1o - this should be in prod now.

**Pull request merged**